### DISCUSSION TITLE
Select scalar from dual with no result return (will we support simple select scalar in the future?)

### DISCUSSION THREAD

**Discussion by ingted** [NTDLS/Katzebase#70] [Q&A]
When having data in table, we could select scalar so far

![image](https://github.com/user-attachments/assets/19a3bae7-2193-4cde-a841-47d25549e672)

However when trying select scalar without physical schema we got problem now:

![image](https://github.com/user-attachments/assets/fdd68c61-4d6b-4074-ac7b-af3d7f44b23f)

or we could use dual (like oracle)

![image](https://github.com/user-attachments/assets/ed37e222-f112-4e58-88ad-0223ef3a0151)

![image](https://github.com/user-attachments/assets/f9bc5e33-c76f-420e-a062-1392a02181e1)

Or what's the correct way to query scalar?

Please take your time to reply all these... stupid... questions (haha... I am still reading the code and try to understand more, PLEASE IGNORE the questions if they are trivial, I know most of the answers would bump out once I get through the whole picture of the project ^____^ )

(I mean I don't want to interrupt your development tempo.)

PS. My timezone is GMT+8

**Comment by NTDLS** [comment]
I’m GMT-4. Yea, I’ve not thought about selecting scalers without a schema. I’ve always thought the “select from dual” was kind of weird, but maybe some “fake” schema like that would work…. Maybe we’ll call it “single”?

oracles “dual” table used to contain 2 rows, but was reduced to 1 and the name was kept for compatibility purposes.

thoughts?

**Reply by NTDLS** [reply]
Also, that error message that just says “dual” is a problem. It should say “undefined schema “dual””.

I’ll fix that.

**Reply by NTDLS** [reply]
Lastly, I pushed a new release today. So be sure to get the latest code. The “big” update I pushed broke a lot of stuff and I’m still finding and fixing issues.

**Comment by NTDLS** [comment]
Added a new system process helper that you can use to run queries from any inside the engine. It is now used for both the login process and the creation of the new "single" schema that is intended to mimic that of Oracle:

```
using var systemSession = Sessions.CreateEphemeralSystemSession();
Documents.InsertDocument(systemSession.Transaction, "Single", "{ephemeral: null}");
systemSession.Commit();
```

Also fixed the processing of math that is in line with string concatenations so now you should be able to execute:
```
select
    11 ^ (2 + 1) + 'ten' + (Length('A10CharStr') * 10 + 2),
    10 + 10 + (11 ^ 3) + 10 + '->' + Guid(),
    10 + 10 + 'ten' + 10 * 10,
    'ten (' + 10 * 10 + ') : ' + DateTimeUTC('yyyy/MM/dd hh:mm:ss tt')
from
    single
```

The Dual thing was a great idea!!!! 💯

**Comment by ingted** [comment]
Resolved in [Release 0.19](https://github.com/NTDLS/NTDLS.Katzebase.Server/releases/tag/0.19.0)
